crates/solvers-dto/src/auction.rs

```diff
@@ -4,61 +4,104 @@ use {
     bigdecimal::BigDecimal,
     number::serialization::HexOrDecimalU256,
     serde::Deserialize,
+    serde_json::{Number, Value},
     serde_with::{serde_as, DisplayFromStr},
     std::collections::HashMap,
+    utoipa::{
+        openapi::{
+            AllOfBuilder,
+            ArrayBuilder,
+            ObjectBuilder,
+            OneOfBuilder,
+            Ref,
+            RefOr,
+            Schema,
+            SchemaType::{self},
+        },
+        ToSchema,
+    },
     web3::types::{H160, H256, U256},
 };
 
+/// The abstract auction to be solved by the searcher.
 #[serde_as]
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, ToSchema)]
 #[serde(rename_all = "camelCase")]
 pub struct Auction {
+    /// An opaque identifier for the auction. Will be set to `null` for requests
+    /// that are not part of an auction (when quoting token prices for example).
     #[serde_as(as = "Option<DisplayFromStr>")]
+    #[schema(value_type = String)]
     pub id: Option<i64>,
-    pub tokens: HashMap<H160, Token>,
+    /// A map of token addresses to token information.
+    pub tokens: HashMap<H160, TokenInfo>,
```

Comment:
isn't there a way to add the value type also the token? the `H160` is type address :thinking: